engine/src/main/java/org/terasology/config/flexible/internal/SettingImpl.java

```diff
@@ -137,13 +143,23 @@ public T getDefaultValue() {
 
     @Override
     public T get() {
+        T systemPropertyValue = getSystemPropertyValue();
+        if (systemPropertyValue != null) {
+            return systemPropertyValue;
+        }
```

Comment:
**thought** Should we have a (private) method to check whether this setting is a system property? This might make the semantics of this method a bit more obvious:

```java
if (isSystemProperty()) {
  return getSystemPropertyValue();
} else {
	return value;
}
```

Is it intended that we return the `value` in case the system property converter yielded a `null `value here?

Reply:
No. System property should rewrite value only.